### PR TITLE
Habit Status functionality added(Assign, enroll, unenroll in date)

### PR DESCRIPTION
**Pull request by TarasBeshlei**

### CODE REVIEW

File: core/src/main/java/greencity/repository/HabitDictionaryRepo.java [hunk 1]
```diff
@@ -18,8 +18,14 @@
      * @param user user which we use to filter.
      * @return List of available {@link HabitDictionary}`s.
      */
-    @Query("SELECT hd FROM HabitDictionary hd WHERE hd.id NOT IN (SELECT h.habitDictionary "
-        + "FROM Habit h WHERE h.user =:user AND h.statusHabit = true)")
+    @Query("SELECT hd "
+        + "FROM HabitDictionary hd "
```
Comment (antonbakalets) [inline core/src/main/java/greencity/repository/HabitDictionaryRepo.java:22]: This must be implemented using JOINs.

Comment (TarasBeshlei) [reply core/src/main/java/greencity/repository/HabitDictionaryRepo.java]: Fixed

File: core/src/main/java/greencity/repository/HabitRepo.java [hunk 2]
```diff
@@ -72,7 +63,14 @@
      * @param end    last day of period
      * @return count of user habits that were marked during some period
      */
-    @Query("SELECT COUNT(h) FROM Habit h WHERE h.user.id = ?1 AND h.statusHabit = true AND h.createDate < ?2 AND h.id "
-        + "IN(SELECT hs.habit.id FROM HabitStatistic hs WHERE hs.createdOn BETWEEN ?2 and ?3)  ")
+    @Query("SELECT u.habits.size "
+        + "FROM User u "
+        + "WHERE u.id = ?1 "
+        + "AND u.habits IN(SELECT h FROM Habit h "
+        + "WHERE h.statusHabit = true "
```
Comment (antonbakalets) [inline core/src/main/java/greencity/repository/HabitRepo.java:70]: I was expecting you to replace all IN statements with JOINs.

Comment (TarasBeshlei) [reply core/src/main/java/greencity/repository/HabitRepo.java]: Sorry, little misunderstanding) Now I fixed all queries.

**Pull request merged**